Deal with SearXNG version

dalf opened the pull request:
Group versions per month:
* 2022-03 (inside this group sort by TLS, CSP and HTML grades, etc...)
* 2022-02 (inside this group, sort by TLS, CSP and HTML grades, etc...)
* 1.0.0
* 0.18.0
* ...

![image](https://user-images.githubusercontent.com/1594191/156426845-153be9e9-1942-4da8-926f-6a199ccd83f3.png)

dalf committed:
Deal with SearXNG version

Group versions per month

unixfox commented:
Thank you :). It would have been better if only the instances that have a version older than 30 days could be in the same group. This way once a new month start the ranking is still "normal" like before the start of the new month.
But this is already a very good step towards a better ranking.

dalf commented:
> It would have been better if only the instances that have a version older than 30 days could be in the same group. 

It requires a visual component to group the versions, otherwise it's going to be confusing.

For reference:
```javascript
const GROUP_VERSION_PER_DAYS = 30;

function normalizeSearxVersion(v) {
    if (typeof (v) !== 'string') {
        return [0, 0, 0, 0, ''];
    }
    const asDate = new Date(v.split("-")[0].replaceAll('.', "-")).getTime();
    if (asDate) {
        // version format "YYYY.MM.DD-HASH" (for example "2022.03.01-0ddcc124")
        // relativeToNow: number of days between now and the release date
        const relativeToNow = Math.round((asDate - VERSION_START_DATE)/(1000*3600*24));
        const relativeToNowRounded = Math.round(relativeToNow / GROUP_VERSION_PER_DAYS);
        const hash = v.split("-")[1];
        return [relativeToNowRounded, 0, 0, 0, hash];
    }
    // version format "MAJOR.MINOR.PATCH-DISTANCE-HASH" (for example "1.0.0-356-c9e6d9f5")
    const vdash = v.split(/[\-\+]/);
    const vdot = vdash[0].split('.').map((i) => parseInt(i, 10));
    if (vdash.length === 1) {
        return [vdot[0], vdot[1], vdot[2], 0, ''];
    }
    if (vdash[1] === 'unknow') {
        return [vdot[0], vdot[1], vdot[2], 0, ''];
    }
    return [vdot[0], vdot[1], vdot[2], parseInt(vdash[1], 10), vdash[2]];
}
```

> But this is already a very good step towards a better ranking.

So I merge this PR and will open another one later with:
* the change above
* an update of the UI
